### ISSUE TITLE
Populate table when copy-pasting tabular data

### ISSUE THREAD

**Issue by ddavidebor** [frappe/erpnext#11912]
This is a small and cute feature request for the UI. Or maybe a bugfix?

```
CC0805JRX7R9104
SMD0805-10K-1%
TC0525B1001T1E
SMD0805-22K-1%
1N5819
SMD2512-1R
FC68125
```

When a text like the above is pasted in "item" field in, for example, a purchase receipt, the text is read as:

`CC0805JRX7R9104 SMD0805-10K-1% TC0525B1001T1E SMD0805-22K-1% 1N5819 SMD2512-1R FC68125`

As you can see the newlines are stripped and substituted with spaces.
When i enter a text with tabs, it is entered with the tabs:

`SMD0805-10K-1%	TC0525B1001T1E	SMD0805-22K-1%	1N5819	SMD2512-1R`

---

This differs from the behavior from the keyboard. When I press "tab" it goes to the next field. When I press "enter" it enters the item date.

It would be extremely helpful in day-to-day use if the UI were to:

* store in memory the pasted string (or a quickly entered string using a barcode reader or simulated keyboard)
* parse it
* take action based on the separators, changing the active field automatically as well as pausing for item data retrieval from the server when needed.
* leave with a red background the fields with problems (for example, no corresponding item found) to be corrected manually.

---

There are A LOT of uses cases for this in day-to-day life. Mostly copy pasting stuff. Examples follow.

Most spreadsheet when copy-pasting cells will output them in the TSV (tab separated values format). With this feature, I could directly paste the following from my supplier exported invoice, without having to use the export-and-import feature.

```
ST-011/R	13
CC0805JRX7R9104	50
SMD0805-10K-1%	25
TC0525B1001T1E	8
SMD0805-22K-1%	75
1N5819	15
SMD2512-1R	75
FC68125	20
16.00M-SMDHC49S	38
CL21C220JBANNNC	50
CL21C220JBANNNC	50
BC846B.215	23
SMD0805-33R-1%	75
SMD0805-56R	75
CL21B105KAFNNNE	40
AD8220ARMZ	2
```

With this, one would be able to not use the import feature for small task. Populating items for purchase receipt, invoices, stock movements etc etc could be done with a simple copy-paste to the standard UI.

---

Basically treating the table where the user enters item-quantity-price (or any similar multi-line input) as a table (which it is)

**Comment by szufisher** [comment]
almost the same feature as I requested 
+1

**Comment by strixaluco** [comment]
I think it is this one: https://github.com/frappe/frappe/issues/4360

**Comment by ddavidebor** [comment]
Yes, more or less the same.

Il lun 11 dic 2017, 09:39 strixaluco <notifications@github.com> ha scritto:

**Comment by tundebabzy** [comment]
In that case, let's close this in favour of frappe/frappe#4360
